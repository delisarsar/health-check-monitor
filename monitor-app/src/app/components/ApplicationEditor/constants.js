import React from "react";
import { gql } from "apollo-boost";
import { FormattedMessage } from "react-intl";

export const APP_NAME_CHARACTER_LIMIT = 255;
export const APP_ENDPOINT_CHARACTER_LIMIT = 255;

export const HEALTH_CHECK_LABEL = (
  <FormattedMessage
    id="overview.health-check"
    defaultMessage={"Health Check"}
  />
);

export const getModalStyle = () => {
  const top = 50;
  const left = 50;

  return {
    top: `${top}%`,
    left: `${left}%`,
    transform: `translate(-${top}%, -${left}%)`,
  };
};

export const CREATE_LABEL = (
  <FormattedMessage id="overview.create" defaultMessage={"Create"} />
);

export const SAVE_LABEL = (
  <FormattedMessage id="overview.save" defaultMessage={"Save"} />
);

export const DELETE_LABEL = (
  <FormattedMessage id="overview.delete" defaultMessage={"Save"} />
);

export const CREATE_APPLICATION = gql`
  mutation CreateApplication($name: String!, $endpoint: String!) {
    createApplication(name: $name, endpoint: $endpoint)
  }
`;

export const DELETE_APPLICATION = gql`
  mutation DeleteApplication($applicationId: Int!) {
    deleteApplication(applicationId: $applicationId)
  }
`;

export const UPDATE_APPLICATION = gql`
  mutation UpdateApplication($name: String!, $endpoint: String!, $appId: Int!) {
    updateApplication(name: $name, endpoint: $endpoint, appId: $appId)
  }
`;

export const GET_APPLICATION_BY_ID = gql`
  query GetApplicationsById($appId: Int) {
    getApplicationById(appId: $appId) {
      id
      name
      healthCheck {
        endpoint
      }
    }
  }
`;
