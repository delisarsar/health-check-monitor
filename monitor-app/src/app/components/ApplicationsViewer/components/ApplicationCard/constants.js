import React from "react";
import { FormattedMessage } from "react-intl";

export const EDIT_APPLICATION_LABEL = (
  <FormattedMessage id="overview.edit" defaultMessage={"Edit"} />
);

export const ENDPOINT_APPLICATION_LABEL = (
  <FormattedMessage id="overview.endpoint" defaultMessage={"Endpoint"} />
);
