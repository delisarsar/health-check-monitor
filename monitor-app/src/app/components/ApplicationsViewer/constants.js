import React from "react";
import { FormattedMessage } from "react-intl";
import { gql } from "apollo-boost";

export const CREATE_APPLICATION_LABEL = (
  <FormattedMessage
    id="features.home.create-application"
    defaultMessage={"+ Create"}
  />
);

export const FILTER_BY_NAME_LABEL = (
  <FormattedMessage
    id="overview.filter-by-name"
    defaultMessage={"Filter by name"}
  />
);

export const NO_APPLICATION_FOUND_LABEL = (
  <FormattedMessage
    id="overview.no-application-found"
    defaultMessage={"No application found"}
  />
);

export const APPLICATIONS_TITLE_LABEL = (
  <FormattedMessage
    id="overview.applications"
    defaultMessage={"Applications (Health Checks)"}
  />
);

export const GET_APPLICATIONS_LIKE_NAME = gql`
  query GetApplicationsLikeName($name: String) {
    getApplicationsLikeName(name: $name) {
      id
      name
      healthCheck {
        endpoint
      }
    }
  }
`;
