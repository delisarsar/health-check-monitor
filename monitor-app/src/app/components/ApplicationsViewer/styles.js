import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  createButton: {
    textTransform: "none",
  },
  resultsTitle: {
    paddingTop: theme.spacing(2),
  },
  appsGrid: {
    paddingTop: theme.spacing(2),
  },
}));

export default useStyles;
