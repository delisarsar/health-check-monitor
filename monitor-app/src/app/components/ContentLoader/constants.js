import React from "react";
import { FormattedMessage } from "react-intl";

export const GENERIC_APP_ERROR = (
  <FormattedMessage
    id="app.error"
    defaultMessage={"An error occured. Please try again"}
  />
);
