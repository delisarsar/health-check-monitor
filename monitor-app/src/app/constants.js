import React from "react";
import { FormattedMessage } from "react-intl";

export const APPLICATION_NAME_LABEL = (
  <FormattedMessage
    id="overview.application-name"
    defaultMessage={"Application name"}
  />
);
