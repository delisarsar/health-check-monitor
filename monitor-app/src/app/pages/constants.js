export const OVERVIEW_PAGE_ROUTE = "/";
export const EDIT_PAGE_ROUTE = "/edit";
export const EDIT_PAGE_APPLICATION_ID_PARAM = "applicationId";
