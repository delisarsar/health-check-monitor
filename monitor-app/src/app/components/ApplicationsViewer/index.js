import React, { useState, useEffect } from "react";
import { useDebounce } from "use-debounce";
import { useQuery } from "@apollo/react-hooks";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import { useIntl } from "react-intl";
import isEmpty from "lodash/isEmpty";
import ApplicationCard from "./components/ApplicationCard";
import ContentLoader from "../ContentLoader";
import { APPLICATION_NAME_LABEL } from "../../constants";
import {
  EDIT_PAGE_ROUTE,
  EDIT_PAGE_APPLICATION_ID_PARAM,
} from "../../pages/constants";
import {
  NO_APPLICATION_FOUND_LABEL,
  APPLICATIONS_TITLE_LABEL,
  CREATE_APPLICATION_LABEL,
  GET_APPLICATIONS_LIKE_NAME,
} from "./constants";
import useStyles from "./styles";
import { useHistory } from "react-router-dom";
import { Grid, Typography } from "@material-ui/core";

const ApplicationsViewer = () => {
  const intl = useIntl();
  const classes = useStyles();
  const history = useHistory();
  const [applicationName, setApplicationName] = useState("");
  const [searchValue] = useDebounce(applicationName, 500);

  const redirectToCreateAppPage = () => {
    history.push(EDIT_PAGE_ROUTE);
  };

  const redirectToEditAppPage = (applicationId) => {
    history.push(
      `${EDIT_PAGE_ROUTE}?${EDIT_PAGE_APPLICATION_ID_PARAM}=${applicationId}`
    );
  };

  const { loading, data, refetch } = useQuery(GET_APPLICATIONS_LIKE_NAME, {
    variables: { name: searchValue },
  });

  useEffect(() => {
    refetch();
  }, [history, refetch, data]);

  return (
    <ContentLoader loading={loading}>
      <div>
        <Grid container spacing={2}>
          <Grid item>
            <Button
              className={classes.createButton}
              data-testid="create-application"
              variant="contained"
              onClick={redirectToCreateAppPage}
            >
              {CREATE_APPLICATION_LABEL}
            </Button>
          </Grid>
          <Grid item>
            <TextField
              label={APPLICATION_NAME_LABEL}
              variant="outlined"
              size="small"
              value={applicationName}
              onChange={(e) => setApplicationName(e.target.value)}
              placeholder={intl.formatMessage({
                id: "overview.filter-by-name",
                defaultMessage: "Filter by name",
              })}
            />
          </Grid>
        </Grid>
        {isEmpty(data) || isEmpty(data.getApplicationsLikeName) ? (
          <Typography
            data-testid="overview-no-applications-found"
            className={classes.resultsTitle}
            variant="h5"
          >
            {NO_APPLICATION_FOUND_LABEL}
          </Typography>
        ) : (
          <>
            <Typography className={classes.resultsTitle} variant="h5">
              {APPLICATIONS_TITLE_LABEL}
            </Typography>
            <Grid container spacing={2} className={classes.appsGrid}>
              {data.getApplicationsLikeName.map((app) => (
                <Grid key={app.id} item>
                  <ApplicationCard
                    id={app.id}
                    appName={app.name}
                    healthCheckEndpoint={app.healthCheck.endpoint}
                    onChange={redirectToEditAppPage}
                  />
                </Grid>
              ))}
            </Grid>
          </>
        )}
      </div>
    </ContentLoader>
  );
};

export default ApplicationsViewer;
