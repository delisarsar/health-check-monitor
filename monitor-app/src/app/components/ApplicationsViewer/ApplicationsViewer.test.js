import React from "react";
import { render, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MockedProvider } from "@apollo/client/testing";
import { GET_APPLICATIONS_LIKE_NAME } from "./constants";
import { IntlProvider } from "react-intl";
import messages from "../../../translations/en.json";
import ApplicationsViewer from ".";
import { Router } from "react-router-dom";

const nonEmptyResponse = [
  {
    request: {
      query: GET_APPLICATIONS_LIKE_NAME,
      variables: {
        name: "",
      },
    },
    result: {
      errors: [],
      data: {
        getApplicationsLikeName: [
          {
            id: 1,
            name: "Jahia Cloud",
            healthCheck: {
              endpoint: "https://jahia.com",
            },
          },
        ],
      },
    },
  },
];

const emptyResponse = [
  {
    request: {
      query: GET_APPLICATIONS_LIKE_NAME,
      variables: {
        name: "",
      },
    },
    result: {
      errors: [],
      data: {
        getApplicationsLikeName: [],
      },
    },
  },
];

describe("<ApplicationsViewer />", () => {
  test("renders correctly with a sample application", async () => {
    const { container, getByTestId } = render(
      <MockedProvider mocks={nonEmptyResponse} addTypename={false}>
        <IntlProvider locale="en-US" messages={messages}>
          <ApplicationsViewer />
        </IntlProvider>
      </MockedProvider>
    );
    expect(getByTestId("progress-loader")).toBeInTheDocument();

    await waitFor(
      () =>
        expect(getByTestId("card-app-name")).toHaveTextContent("Jahia Cloud"),
      () => expect(getByTestId("card-endpoint")).toContain("https://jahia.com")
    );

    expect(container).toMatchSnapshot();
  });

  test("renders a no applications label when it's empty", async () => {
    const { container, getByTestId } = render(
      <MockedProvider mocks={emptyResponse}>
        <IntlProvider locale="en-US" messages={messages}>
          <ApplicationsViewer />
        </IntlProvider>
      </MockedProvider>
    );
    expect(getByTestId("progress-loader")).toBeInTheDocument();

    await waitFor(() =>
      expect(getByTestId("overview-no-applications-found")).toBeInTheDocument()
    );

    expect(container).toMatchSnapshot();
  });

  test("redirect to the create new application page when the button is clicked ", async () => {
    const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
    const { getByTestId } = render(
      <Router history={historyMock}>
        <MockedProvider mocks={nonEmptyResponse}>
          <IntlProvider locale="en-US" messages={messages}>
            <ApplicationsViewer />
          </IntlProvider>
        </MockedProvider>
      </Router>
    );
    expect(getByTestId("progress-loader")).toBeInTheDocument();

    await waitFor(() =>
      expect(getByTestId("create-application")).toBeInTheDocument()
    );

    userEvent.click(getByTestId("create-application"));

    expect(historyMock.push).toHaveBeenCalledWith("/edit");
  });

  test("redirects to the edit page route in order to edit a card ", async () => {
    const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
    const { getByTestId } = render(
      <Router history={historyMock}>
        <MockedProvider mocks={nonEmptyResponse} addTypename={false}>
          <IntlProvider locale="en-US" messages={messages}>
            <ApplicationsViewer />
          </IntlProvider>
        </MockedProvider>
      </Router>
    );

    expect(getByTestId("progress-loader")).toBeInTheDocument();
    await waitFor(() =>
      expect(getByTestId("application-card-1")).toBeInTheDocument()
    );

    userEvent.click(getByTestId("edit-button"));
    expect(historyMock.push).toHaveBeenCalledWith("/edit?applicationId=1");
  });
});
