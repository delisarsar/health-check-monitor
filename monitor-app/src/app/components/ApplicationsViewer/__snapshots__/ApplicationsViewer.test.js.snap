// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplicationsViewer /> renders a no applications label when it's empty 1`] = `
<div>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-createButton-20"
          data-testid="create-application"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            + Create
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
          >
            Application Name
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              placeholder="Filter by name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-27 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-29"
              >
                <span>
                  Application Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <h5
      class="MuiTypography-root makeStyles-resultsTitle-21 MuiTypography-h5"
      data-testid="overview-no-applications-found"
    >
      No application found
    </h5>
  </div>
</div>
`;

exports[`<ApplicationsViewer /> renders correctly with a sample application 1`] = `
<div>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-createButton-1"
          data-testid="create-application"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            + Create
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
          >
            Application Name
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              placeholder="Filter by name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-14"
              >
                <span>
                  Application Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <h5
      class="MuiTypography-root makeStyles-resultsTitle-2 MuiTypography-h5"
    >
      Applications (Health Checks)
    </h5>
    <div
      class="MuiGrid-root makeStyles-appsGrid-3 MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiPaper-root MuiCard-root makeStyles-root-16 MuiPaper-elevation2 MuiPaper-rounded"
          data-testid="application-card-1"
        >
          <div
            class="MuiCardContent-root makeStyles-cardContent-19"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
              data-testid="card-app-name"
            >
              Jahia Cloud
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
              data-testid="card-endpoint"
            >
              Endpoint
              : 
              https://jahia.com
            </p>
          </div>
          <div
            class="MuiCardActions-root makeStyles-cardActions-18 MuiCardActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text"
              data-testid="edit-button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Edit
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
